### PR TITLE
School library: set up associations

### PR DESCRIPTION
**Pull request by Adams-Ghub**
The following were carried out 

- Created rental, book, classroom and student classes
- Establish the various relationship among the various classes
- Check and correct linter errors

### CODE REVIEW

File: rental.rb [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+  end
+
+  def book=(book)
+    @book = book
+    book.rental << self unless book.rental.include?(self)
+  end
+
+  def person=(person)
+    person.rental << self unless person.rental.include?(self)
+  end
```
Comment (anagudelogu) [inline rental.rb]: - [ ] You did a great job working on this milestone! 💯 However, this doesn't work as you could expect:

When you create a new `Rental`, you want that instance to be added to each `Book` and `Person` rentals array. However, this is not happening in your code, the setter methods `book=` and `person=` doesn't get called on initialization.

For this to work, you would have to use those methods inside your initialize method:

```rb
def initialize(date, person, book)
  @date = date
  self.person = person
  self.book = book
end
```

Which doesn't feel that natural. Instead, I would suggest you merge those methods with the initialize one, ending up with something like this:

```suggestion
  def initialize(date, person, book)
    @date = date
    @person = person
    person.rental << self unless person.rental.include?(self)
    @book = book
    book.rental << self unless book.rental.include?(self)
  end
  
  
  
  
  
  
  
```

Please consider implementing this change to apply the both-ways relationship concept correctly! 🤓 🔧

**Review by anagudelogu — CHANGES_REQUESTED** [review]
# STATUS:  CHANGES REQUIRED  ♻️ 🚧 

Hi @Adams-Ghub, 
I'm Andres.


While you have done a great job working on this milestone 💻 , there are some issues that you still need to work on to go to the next project. ♻️ 

You are almost there, keep up the good work! 🚀 



## Highlights
✅ Great use of Git flow. 💯 
✅ Your project is documented professionally. 📜 
✅ Descriptive PR title, summary and commit messages. 💬 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: rental.rb (hunk 1 above)
Comment (Adams-Ghub) [reply rental.rb]: Thank you so much for the suggestion

**Review by DammyShittu — APPROVED** [review]
Hi @Adams-Ghub,

## APPROVED 🟢

## Highlights
✔️ Associations

Great implementation of all the requirements in this milestone. 👏 

_You have done an amazing job 👍_

Your project is complete!  There is nothing else to say other than... it's time to merge it. :shipit: 

Congratulations! 🎉 

## CHEERS AND HAPPY CODING!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**